### PR TITLE
Fix late initialization error for `Analytics.userProperty`

### PR DESCRIPTION
**Pull request by eliasyishak**
Fixes:
- https://github.com/dart-lang/tools/issues/238

TLDR: initialization logic within the `Session` class's constructor body expects `Analytics.userProperty` to be initialized, this moves that constructor body logic from `Session` into a separate method (`Session.initialize`) we will call at the end of the `Analytics` class's constructor after we are sure all necessary fields have been initialized

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/unified_analytics/lib/src/session.dart [hunk 1]
```diff
@@ -70,6 +63,16 @@ class Session {
     return _sessionId;
   }
 
+  /// Preps the [Session] class with the data found in the session json file.
+  ///
+  /// We must check if telemetry is enabled to refresh the session data
+  /// because the refresh method will write to the session file and for
+  /// users that have opted out, we have to leave the session file empty
+  /// per the privacy document
+  void initialize(bool telemetryEnabled) {
```
Comment (christopherfujino) [inline pkgs/unified_analytics/lib/src/session.dart:72]: it worries me anytime there is an `initialize()` function that one must remember to call at the right time, because a refactor could easily lead to code now accessing state in this class BEFORE the `initialize()` method was called.

I think this would solve the particular issue linked, and maybe this is the right change to make immediately to stop crashes, but I think long-term we should refactor all of these classes (especially `Analytics`) such that it is not possible to access any of its state BEFORE it has been initialized.

**Review by christopherfujino — APPROVED** [review]
This LGTM as a fix to get to users.

As discussed, I think ideally we would make all class fields either `final` or `late final` with initializer expressions, to ensure we don't hit LateInitializationErrors.

**Comment by eliasyishak** [comment]
Going to merge this in now and publish so we can have a fix, filed a separate issue to refactor session handler
- https://github.com/dart-lang/tools/issues/240

**Comment by christopherfujino** [comment]
> Going to merge this in now and publish so we can have a fix, filed a separate issue to refactor session handler
> 
> * [Refactor analytics session handler to use `File.lastModifiedSync` for `_lastPing` #240](https://github.com/dart-lang/tools/issues/240)

Can we also refactor all the late/late final fields to something safer? I feel like unless one can reason globally about all the state in this package and all possible edge cases, it's impossible to know that users won't hit another LateInitializationError.

**Comment by eliasyishak** [comment]
Definitely, I captured here: https://github.com/dart-lang/tools/issues/241

**Comment by christopherfujino** [comment]
> Definitely, I captured here: #241

nice, thanks!
